src/ios/BranchSDK.m

```diff
@@ -84,31 +84,30 @@ - (void)initSession:(CDVInvokedUrlCommand*)command
         BOOL isFromUniversalLink = NO;
         NSString *resultString = nil;
         CDVPluginResult *pluginResult = nil;
-
+        
         // NOTE: For Universal Links. Using clicked_branch_link key as condition at the moment to identify if block is run due to Universal Links.
         isFromUniversalLink = [[params objectForKey:@"+clicked_branch_link"] boolValue];
-
+        
         if (!error) {
-            if (params != nil && [params count] > 0) {
+            if (params != nil && [params count] > 0 && isFromUniversalLink) {
                 NSLog(@"Success");
-                pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:params];
-                if (isFromUniversalLink) {
-                    NSError *err;
-                    NSData *jsonData = [NSJSONSerialization dataWithJSONObject:@{@"params":params}
-                                                                       options:0
-                                                                         error:&err];
-                    if (!jsonData) {
-                        NSLog(@"Parsing Error: %@", [err localizedDescription]);
-                        NSDictionary *errorDict = [NSDictionary dictionaryWithObjectsAndKeys:[err localizedDescription], @"error", nil];
-                        NSData* errorJSON = [NSJSONSerialization dataWithJSONObject:errorDict
-                                                                            options:NSJSONWritingPrettyPrinted
-                                                                              error:&err];
-
-                        resultString = [[NSString alloc] initWithData:errorJSON encoding:NSUTF8StringEncoding];
-                    } else {
-                        NSLog(@"Success");
-                        resultString = [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
-                    }
+                NSError *err;
+                NSData *jsonData = [NSJSONSerialization dataWithJSONObject:params
```

Comment:
In success below, I think you want to do messageAsDictionary:params. That should make it consistent with Java. Right now the code returns { params: params }, which is nice, but inconsistent. We didn't need { params: params } so I removed that level of nesting, and I think we should always return objects as we do in Java (I believe the current code returns JSON?).